lstchain/io/io.py

```diff
@@ -1264,37 +1261,32 @@ def remove_duplicated_events(data):
     data.remove_rows(remove_row_list)
 
 
-def parse_cfg_bytestring(bytestring):
-    """
-    Parse configuration as read by eventio
-    :param bytes bytestring: A ``Bytes`` object with configuration data for one parameter
-    :return: Tuple in form ``('parameter_name', 'value')``
-    """
-    line_decoded = bytestring.decode('utf-8').rstrip()
-    if 'ECHO' in line_decoded or '#' in line_decoded:
-        return None
-    line_list = line_decoded.split('%', 1)[0]  # drop comment
-    res = re.sub(' +', ' ', line_list).strip().split(' ', 1)  # remove extra whitespaces and split
-    return res[0].upper(), res[1]
-
-
 def extract_simulation_nsb(filename):
     """
     Get current run NSB from configuration in simtel file
     :param str filename: Input file name
-    :return array of `float` by tel_id: NSB rate
-    """
-    nsb = []
-    with EventIOFile(filename) as f:
-        for o in yield_all_subobjects(f, [History, HistoryConfig]):
-            if hasattr(o, 'parse'):
-                try:
-                    cfg_element = parse_cfg_bytestring(o.parse()[1])
-                    if cfg_element is not None:
-                        if cfg_element[0] == 'NIGHTSKY_BACKGROUND':
-                            nsb.append(float(cfg_element[1].strip('all:')))
-                except Exception as e:
-                    print('Unexpected end of %s,\n caught exception %s', filename, e)
+    :return dict of `float` by tel_id: NSB rate
+    """
+    nsb = {}
+    # In current MC, correct NSB are logged after 'STORE_PHOTOELECTRONS' entries
+    # TODO In any new production, behaviour needs to be verified.
+    # New version of simtel will allow to use better metadata
+    next_nsb = False
+    tel_id = 1
+    with SimTelFile(filename) as f:
+        try:
+            for _, line in f.history:
```

Comment:
This seems very brittle. Don't we have a better way to understand which option is the correct one to use other than the one that follow `STORE_PHOTOELECTRONS`?

Reply:
I don't see one. But I am not an expert of simtel so maybe I am missing something. As far as I know the information is only located here and there is no clear indication of its usage.

What I checked is that, for whatever reason, the nsb entries corresponding to the final config of each telescope follow this pattern for the two latest MC productions (in the files I checked). But not other entries.

I agree it is brittle and the use of the new metadata available in Simtel will be better once we use it for MC productions. Until then, we have to hope that the config used are somewhat consistent...

Reply:
I see no better option as of now. 
Perhaps you can add a clear warning log message stating what value will be used, so that the user can eventually check on the simtel config files that it is what it should be.

Reply:
Good idea, this info is currently given in a info level message without warning the user to check it. I will add a dedicated warning message.